pkg/reconciler/instances/istio/action.go

```diff
@@ -1,410 +1,132 @@
+		if err != nil {
+			return err
 		}
 	}
-	return nil
-}
-
-func dataPlaneVersionsString(istioStatus actions.IstioStatus, delimiter string) string {
-	dpVersions := []string{}
-	for version := range istioStatus.DataPlaneVersions {
-		dpVersions = append(dpVersions, version)
-	}
-	return strings.Join(dpVersions[:], delimiter)
-}
-
-func isComponentCompatible(componentVersion, targetVersion, componentName string) (bool, error) {
-	if componentVersion == "" {
-		return true, nil
-	}
-	componentHelperVersion, err := newHelperVersionFrom(componentVersion)
-	if err != nil {
-		return false, err
-	}
-	targetHelperVersion, err := newHelperVersionFrom(targetVersion)
-	if err != nil {
-		return false, err
-	}
 
-	componentVsTargetComparison := targetHelperVersion.compare(componentHelperVersion)
-	if !amongOneMinor(componentHelperVersion, targetHelperVersion) {
-		return false, fmt.Errorf("could not perform %s for %s from version: %s to version: %s - the difference between versions exceed one minor version",
-			getActionTypeFrom(componentVsTargetComparison), componentName, componentVersion, targetVersion)
-	}
-
-	return true, nil
-}
-
-func getActionTypeFrom(comparison int) string {
-	switch comparison {
-	case 1:
-		return "upgrade"
-	case 0:
-		return "reconciliation"
-	case -1:
-		return "downgrade"
-	default:
-		return "unknown"
-	}
-}
-
-func amongOneMinor(first, second helperVersion) bool {
-	return first.ver.Major == second.ver.Major && (first.ver.Minor == second.ver.Minor || first.ver.Minor-second.ver.Minor == -1 || first.ver.Minor-second.ver.Minor == 1)
-}
-
-func unDeployIstioRelatedResources(context context.Context, manifest string, client kubernetes.Client, logger *zap.SugaredLogger) error {
-	logger.Debugf("Undeploying istio related dashboards")
-	// multiple calls necessary, please see: https://github.com/kyma-incubator/reconciler/issues/367
-	_, err := client.Delete(context, manifest, "kyma-system")
-	if err != nil {
-		return err
-	}
-	logger.Debugf("Undeploying other istio related resources")
-	_, err = client.Delete(context, manifest, istioNamespace)
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
-
-type envoyAction struct {
-	getIstioPerformer bootstrapIstioPerformer
+	return retry.Do(func() error {
+		return checkIfIstioIsReady(context, k8sClient)
+	}, retry.Attempts(retryAttempts), retry.Delay(retryDelay))
 }
 
-func NewEnvoyAction(getIstioPerformer bootstrapIstioPerformer) service.Action {
-	return &envoyAction{
-		getIstioPerformer: getIstioPerformer,
-	}
-}
-
-func (e *envoyAction) Run(context *service.ActionContext) error {
-	context.Logger.Debug("Istio envoy filter apply action triggered")
-
-	provider := clientset.DefaultProvider{}
-
-	istioClient, err := provider.GetIstioClient(context.KubeClient.Kubeconfig())
+func checkIfIstioIsReady(context *service.ActionContext, k8sClient client.Client) error {
+	var istioList v1alpha1.IstioList
+	err := k8sClient.List(context.Context, &istioList)
```

Comment:
is we want to provide valid feedback if Istio is ready, we should take into account istio operator reconcile for oldest Istio CR in `kyma-system`. You should utilize: https://github.com/kyma-project/istio/blob/a6070ecf8c4c21356c1a8f5400162f4b6e0eb1eb/controllers/istio_controller.go#L245

we do not care for other Istio CRs (other namespace or not the oldest one).